File: LoggingCS/FileLogger/FileLoggerOptionsSetup.cs
using Microsoft.Extensions.Logging.Configuration;
using Microsoft.Extensions.Options;

namespace ova.Common.Logging.FileLogger
{
    internal class FileLoggerOptionsSetup : ConfigureFromConfigurationOptions<FileLoggerOptions>
    {
        public FileLoggerOptionsSetup(ILoggerProviderConfiguration<FileLoggerProvider>providerConfiguration): base(providerConfiguration.Configuration) { }
    }
}

File: Logging/FileLogging/FileLogging/FileLogger/FileLoggerOptionsSetup.vb
Imports Microsoft.Extensions.Logging.Configuration
Imports Microsoft.Extensions.Options

Namespace FileLogger
    Friend Class FileLoggerOptionsSetup
        Inherits ConfigureFromConfigurationOptions(Of FileLoggerOptions)

        Public Sub New(ByVal providerConfiguration As ILoggerProviderConfiguration(Of FileLoggerProvider))
            MyBase.New(providerConfiguration.Configuration)
        End Sub
    End Class
End Namespace
